lib/iris/pandas.py

```diff
@@ -129,6 +129,9 @@ def _assert_shared(np_obj, pandas_obj):
         base = pandas_obj.base
     else:
         base = pandas_obj[0].base
+
+    if pandas_obj.values is np_obj:
+        return
```

Comment:
please add a comment around this change, explaining why the change is needed and referencing pandas versions as appropriate

I would like to see just enough about the pandas implementation change to be able to understand the change if I came back to it